### PR TITLE
feat: initial basic footer

### PR DESCRIPTION
**Pull request by abutterworth**
Usage
-----

To install frontend-component-footer-edx into your project
```
   npm i --save @edx/frontend-component-footer-edx
```
Component Usage
```
   import Footer from '@edx/frontend-component-footer-edx';
   import footerMessages from '@edx/frontend-component-footer-edx/src/i18n/index';

   ...

   <Footer
      handleAllTrackEvents={(eventName, properties) => {/* track click event */}}
      onLanguageSelected={(languageCode) => {/* set language */}}
      supportedLanguages={[
         { label: 'English', value: 'en'},
         { label: 'Español', value: 'es' },
      ]}
   />
```
- handleAllTrackEvents (required)
- onLanguageSelected (optional)
- supportedLanguages (optional)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+install:
+  - npm install
+script:
+  - npm run lint
+  - npm run i18n_extract
+  - npm run test
+  - npm run build
+after_success:
+  - npx semantic-release
+  - npm run coveralls
```
Comment (davidjoy) [inline .travis.yml]: Just confirmed with @estute that we're trying to move to codecov instead of coveralls.

File: example/index.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { render } from 'react-dom';
+import { IntlProvider } from '@edx/frontend-i18n';
+
+import './index.scss';
+import Footer from '../src';
+
+const App = () => (
+  <div>
+    <IntlProvider locale="en">
```
Comment (davidjoy) [inline example/index.jsx]: For what it's worth, you should be able to use frontend-base in this example app, like the header repos. (I don't feel strongly)

File: package.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  "description": "Site footer component for use when building edX frontend applications",
+  "main": "dist/index.js",
+  "publishConfig": {
+    "access": "public"
+  },
+  "scripts": {
+    "build": "make build",
+    "gc": "commit",
+    "commitmsg": "commitlint -e $GIT_PARAMS",
+    "coveralls": "cat ./coverage/lcov.info | coveralls",
```
Comment (davidjoy) [inline package.json]: codecooooov

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import Footer, { EVENT_NAMES } from './components/Footer';
```
Comment (davidjoy) [inline src/index.js]: This file can read:

```
export { default, EVENT_NAMES } from './components/Footer';
export { default as messages } from './i18n/index';
```

File: example/index.jsx (hunk 2 above)
Comment (abutterworth) [reply example/index.jsx]: I figure this is just as easy.

**Pull request merged**